core/src/main/java/com/graphhopper/GraphHopper.java

```diff
@@ -909,32 +918,35 @@ public boolean load() {
-            if (properties.containsVersion())
-                throw new IllegalStateException("The GraphHopper file format is not compatible with the data you are " +
-                        "trying to load. You either need to use an older version of GraphHopper or run a new import");
-            // check encoding for compatibility
-            String flagEncodersStr = properties.get("graph.flag_encoders");
+    private void loadEncodingManagerFromProperties(StorableProperties properties) {
+        if (properties.containsVersion())
+            throw new IllegalStateException("The GraphHopper file format is not compatible with the data you are " +
+                    "trying to load. You either need to use an older version of GraphHopper or run a new import");
+        String encodedValueStr = properties.get("graph.encoded_values");
+        ArrayNode evList = deserializeEncodedValueList(encodedValueStr);
+        List<EncodedValue> encodedValues = new ArrayList<>();
+        evList.forEach(serializedEV -> encodedValues.add(EncodedValueSerializer.deserializeEncodedValue(serializedEV.textValue())));
+
+        String flagEncodersStr = properties.get("graph.flag_encoders");
+        List<FlagEncoder> flagEncoders = Stream.of(flagEncodersStr.split(","))
+                .map(str -> flagEncoderFactory.deserializeFlagEncoder(str, name -> encodedValues.stream()
+                        .filter(ev -> ev.getName().equals(name))
+                        .findFirst()
+                        .orElseThrow(() -> new IllegalStateException("FlagEncoder " + str + " uses unknown encoded value: " + name))))
+                .collect(Collectors.toList());
 
-            if (!encodingManager.toFlagEncodersAsString().equalsIgnoreCase(flagEncodersStr)) {
-                throw new IllegalStateException("Flag encoders do not match:"
-                        + "\nGraphhopper config: " + encodingManager.toFlagEncodersAsString()
-                        + "\nLoaded graph:       " + flagEncodersStr
-                        + "\nChange configuration to match the graph or delete " + baseGraph.getDirectory().getLocation());
-            }
+        EncodingManager.Builder emBuilder = EncodingManager.start();
+        flagEncoders.forEach(emBuilder::addWithoutAddingEncodedValues);
+        encodedValues.forEach(emBuilder::add);
+        encodingManager = emBuilder.build();
+    }
 
-            String encodedValueStr = properties.get("graph.encoded_values");
-            if (!encodingManager.toEncodedValuesAsString().equalsIgnoreCase(encodedValueStr)) {
-                throw new IllegalStateException("Encoded values do not match:"
-                        + "\nGraphhopper config: " + encodingManager.toEncodedValuesAsString()
-                        + "\nLoaded graph:       " + encodedValueStr
-                        + "\nChange configuration to match the graph or delete " + baseGraph.getDirectory().getLocation());
-            }
```

Comment:
This is the code that was forcing us to pass the same configuration when loading the graph even though it was already determined by the import.